build/accelerate_launch.py

```diff
@@ -80,6 +77,21 @@ def main():
             or 'SFT_TRAINER_CONFIG_JSON_ENV_VAR'."
             )
 
+        # Configure log_level of python native logger.
+        log_level = job_config.get(
+            "log_level"
+        )  # this will be set to either the value found or None
+        if (
+            not log_level
+        ):  # if log level not set by job_config aka by JSON, set it via env var or set default
+            log_level = os.environ.get("LOG_LEVEL", "WARNING")
+        log_level = log_level.upper()
+        logging.basicConfig(level=log_level)
+
+        # Configure for Image to get log level as the code after
+        # launch_command only takes log level from env var LOG_LEVEL and not CLI
+        os.environ["LOG_LEVEL"] = log_level
```

Comment:
technically you shouldn't set the env LOG_LEVEL if the user already set it. so something more like:
```py
log_level = job_config.get("log_level")
if not log_level:
   log_level = os.environ.get("LOG_LEVEL", "WARNING")
else:
   os.environ["LOG_LEVEL"] = log_level.upper()
logging.basicConfig(level=log_level.upper())
```

Reply:
Actually, I wrote the same code what you mentioned here, but not sure why, a weird thing was happening last week when running this log level logic in Image:

Whenever `sfttrainer.py` is run with accelerate, with CLI argument `--log_level` it doesn't take in this cli argument when executing `launch_command(args)` and hence  `log_level` CLI argument doesn't reach `sfttrainer.py`. But if I set log level by env variable, it consider that and runs fine as per logic. 

Hence as a workaround I check CLI value in `accelerate_launch.py` along with env variable and accordingly as per logic assigns the log_level to "LOG_LEVEL" env var so that `launch_command(args)` takes this env var and runs with correct log level.

Reply:
> Whenever sfttrainer.py is run with accelerate, with CLI argument --log_level it doesn't take in this cli argument when executing launch_command(args) 

The accelerate launch script only parses JSON to and not the flags. So passing `--log_level` would never work for accelerate launch but passing it in via JSON `"log_level": "info"` should. Was this also not working?

> log_level CLI argument doesn't reach sfttrainer.py

Please describe, if you pass in log_level via JSON it should be interpreted both by accelerate_launch and sft_trainer scripts.

And if the user sets `export LOG_LEVEL=info` this should also be interpreted both by accelerate_launch and sft_trainer scripts. 

The logic I have above is not too different frmo yours, it just doesn't reset the env var LOG_LEVEL if the user already sets this value.

Reply:
> The accelerate launch script only parses JSON to and not the flags. So passing --log_level would never work for accelerate launch but passing it in via JSON "log_level": "info" should. Was this also not working?

Yes, it was not taking the config json value of `--log_level`. I will share the example with you shortly.

Reply:
Okay, I check again and it's working. Not sure why it was not taking in leg level from json file last week. 
Thanks though!